src/chapters/style/documentation.typ

```diff
@@ -0,0 +1,247 @@
+    ...
+  ]
+  ```
+]
+
+Doc comments contain a description of the documented item itself, as well as an optional semantic trailer.
+The content of descriptions should generally be simple Typst markup and should not contain any scripting, (i.e. no loops, conditionals or function calls, except for `#link("...")[...]`), this allows the documentation to be turned into markdown or plain text for language servers to send to editors.
+
+== Description
+As mentioned before, the description should be simple, containing mostly markup and no scripting.
+
+== Semantic Trailer
+The semantic trailer is fully optional, starts with an empty doc comment line to separate it from the description, and may contain:
+- multiple `term` items for parameter type hints and descriptions,
+- a return type hint `-> type`,
+- and multiple property annotations (`#property("private")` or `#property("deprecated")`).
+
+Types in type lists or return type annotations may be separated by `|` to indicate that more than one type is accepted, the exact types allowed depend on the doc parser, but the built in types are generally supported.
+
+Parameter description and return types (if present) are placed tightly together, property annotations if present are separated using another empty doc comment line.
+
+Parameter documentation is created by writing a term item containing the parameter name, a mandatory type list in parentheses and optional description.
+If the parameter is an argument sink its name must likewise contain the spread operator `..`.
+Each parameter can only be documented once, but doesn't have to, undocumented parameters are considered private by convention and may only optional parameters.
+
+#do-dont[
+  ```typst
+  /// Function description
+  ///
+  /// / b (any): b descpription
+  /// / a (int | float): a description
+  #let func(a, b, c) = { ... }
+
+  /// Function description
+  ///
+  /// / ..args (int): args description
+  #let func(..args) = { ... }
+  ```
+][
+  ```typst
+  /// Function description
+  ///
+  /// / c (any): c doesn't exist
+  /// / a (int | float): a description
+  #let func(a, b) = { ... }
+
+  /// Missing empty line between function description and trailer
+  /// / b (any): b descpription
+  /// / a (int | float): a description
+  #let func(a, b) = { ... }
+
+
+  /// Function description
+  ///
+  /// / b (any): b descpription
+  /// / a: missing types
+  #let func(a, b) = { ... }
+
+  /// Function description
+  ///
+  /// / args (int): missing spread operator for args
+  #let func(..args) = { ... }
+  ```
+]
+
+The return type can only be annotated once, on a single line after all parameters, if any exist.
+For non function types the return type annotation can be used as a normal type annotation.
+Items which bind a `function.with` expression should be treated as regular function definitions, i.e. their return type is the return type of the function when called, not `function` itself as would be the return value of `function.with`.
```

Comment:
function.with() ?

Reply:
and comma after i.e.

Reply:
> function.with() ?

The example below explains this, shorthand declarations with partially bound parameters.

Reply:
I meant adding the parentheses 😂

Reply:
Oh, yes sure.